mat64/dense_arithmetic.go

```diff
@@ -124,12 +124,21 @@ func (m *Dense) Norm(ord float64) float64 {
 func (m *Dense) Add(a, b Matrix) {
 	ar, ac := a.Dims()
 	br, bc := b.Dims()
-
 	if ar != br || ac != bc {
 		panic(ErrShape)
 	}
 
+	aMat, _ := untranspose(a)
+	bMat, _ := untranspose(b)
 	m.reuseAs(ar, ac)
+	var f func()
+	m, f = uniqueMemoryDense(m, aMat)
+	if f == nil {
```

Comment:
Okay. I was trying to make a minimal resuable set, but I see your point. I think what you're suggesting is:
````
m.reusaAs(ar,as)
var f func()
if m == a {
    m, f = uniqueMemory(m, aMat)
    defer f()
} else if m == b {
    m, f = uniqueMemory(m, bMat)
    defer f()
}
````
Correct?